packages/common/src/ui/layout/tables/RemoteDataTable.tsx

```diff
@@ -76,39 +70,33 @@ export const RemoteDataTable = <T extends Record<string, unknown>>({
     >
       <MuiTable sx={{ display: 'block', overflow: 'auto' }}>
         <TableHead sx={{ flex: 1, display: 'flex', flexDirection: 'column' }}>
-          {headerGroups.map(({ getHeaderGroupProps, headers }) => (
-            <HeaderRow key={getHeaderGroupProps().key}>
-              {headers.map(column => (
-                <HeaderCell
-                  minWidth={column.minWidth}
-                  width={Number(column.width)}
-                  onSortBy={onSortBy}
-                  key={column.getHeaderProps().key}
-                  isSortable={!column.disableSortBy}
-                  isSorted={column.id === sortBy.key}
-                  align={column.align}
-                  id={column.id as KeyOf<T>}
-                  direction={sortBy.direction}
-                >
-                  {column.render('Header')}
-                </HeaderCell>
-              ))}
-            </HeaderRow>
-          ))}
+          <HeaderRow>
+            {columns.map(column => (
+              <HeaderCell
+                minWidth={column.minWidth}
+                width={Number(column.width)}
+                onSortBy={onSortBy}
+                key={String(column.key)}
+                isSortable={!!column.sortable}
+                isSorted={column.key === sortBy.key}
+                align={column.align}
+                columnKey={column.key}
+                direction={sortBy.direction}
+              >
+                <column.Header column={column} />
+              </HeaderCell>
+            ))}
+          </HeaderRow>
         </TableHead>
         <TableBody sx={{ flex: 1, display: 'flex', flexDirection: 'column' }}>
-          {rows.map((row: Row<T>) => {
-            prepareRow(row);
-
-            const { cells, original } = row;
-            const { key } = row.getRowProps();
-
+          {data.map((row, idx) => {
             return (
               <DataRow<T>
-                cells={cells}
-                key={key}
+                columns={columns}
+                key={row.id}
                 onClick={onRowClick}
-                rowData={original}
+                rowData={row}
+                rowKey={String(idx)}
```

Comment:
passing the `columns` over `Cells` means we can memoise the data row now if we wanted and also just made typing things a lot easier for me..